app/assets/javascripts/utils/course_utils.coffee

```diff
@@ -17,8 +20,12 @@ class CourseUtils
 
   # This builds i18n interface strings that vary based on state/props.
   i18n: (message_key, prefix, default_prefix = 'courses') ->
-    key_prefix = prefix
-    key_prefix ||= default_prefix
-    I18n.t("#{key_prefix}.#{message_key}")
+    if prefix
+      message = I18n.t("#{prefix}.#{message_key}")
+
+    if !message || message.startsWith("[missing")
```

Comment:
This uses a js library, i18n.js, but it's got pretty much the same
interface as the rails i18n framework.

On Monday, February 22, 2016, AndrewGreen <notifications@github.com> wrote:

> In app/assets/javascripts/utils/course_utils.coffee
> <https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/662#discussion_r53673115>
> :
>
> > @@ -17,8 +20,12 @@ class CourseUtils
> >
> >    # This builds i18n interface strings that vary based on state/props.
> >    i18n: (message_key, prefix, default_prefix = 'courses') ->
> > -    key_prefix = prefix
> > -    key_prefix ||= default_prefix
> > -    I18n.t("#{key_prefix}.#{message_key}")
> > +    if prefix
> > +      message = I18n.t("#{prefix}.#{message_key}")
> > +
> > +    if !message || message.startsWith("[missing")
>
> Is this right?
> http://apidock.com/rails/ActionView/Helpers/TranslationHelper/translate
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/662/files#r53673115>
> .
>